scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScaladocParser.scala

```diff
@@ -130,13 +134,73 @@ object ScaladocParser {
     P(startOrNl ~ listParser)
   }
 
+  private val tableParser: Parser[Table] = {
+    def toRow(x: Iterable[String]): Table.Row = Table.Row(x.toSeq)
+    def toAlign(x: String): Option[Table.Align] = {
+      def isEnd(y: Char) = y match {
+        case ':' => Some(true)
+        case '-' => Some(false)
+        case _ => None
+      }
+      for {
+        isLeft <- isEnd(x.head)
+        isRight <- isEnd(x.last)
+        if x.slice(1, x.length - 1).forall(_ == '-')
+      } yield {
+        if (!isRight) Table.Left
+        else if (!isLeft) Table.Right
+        else Table.Center
+      }
+    }
+
+    val cellChar = escape ~ AnyChar | !(nl | escape | tableSep) ~ AnyChar
+    val row = (cellChar.rep.! ~ tableSpaceSep).rep(1)
+    // non-standard but frequently used delimiter line, e.g.: +-----+-------+
+    val delimLine = hspaces0 ~ CharsWhileIn("+-")
+    val sep = nl ~ (delimLine ~ nl).rep ~ tableSpaceSep
+    val table = row.rep(2, sep = sep).map { x =>
+      val rest = x.tail.map(_.map(_.trim))
```

Comment:
Do we know `x` is always non-empty?

Reply:
i will add a test but if i read the `.rep(2)` above correctly, it should have at least 2 elements.